pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -241,7 +241,7 @@ func ResolveTaskRuns(getTaskRun GetTaskRun, state PipelineRunState) error {
 
 // getTaskRunName should return a uniquie name for a `TaskRun`.
 func getTaskRunName(taskRunsStatus map[string]v1alpha1.TaskRunStatus, prName string, pt *v1alpha1.PipelineTask) string {
-	base := fmt.Sprintf("%s-%s", prName, pt.Name)
+	base := names.SimpleNameGenerator.RestrictLengthWithSpaceForSuffix(fmt.Sprintf("%s-%s", prName, pt.Name))
```

Comment:
I thought about just exposing `maxGeneratedNameLength` and doing `fmt.Sprintf("%s-%s", prName, pt.name)[:names.MaxGeneratedNameLength]`, but opted to go this route for consistency. The name is kinda terrible though. =)